Lab2/lab2_drones_manager.cpp

```diff
@@ -109,13 +109,12 @@ bool DronesManager::insert_front(DroneRecord value) {
         size++;
         return true;
     } else {
-        value.next = first->next;
-        first = &value;
-        if (value.next != NULL) {
-            value.next->prev = &value;
-        }
-        first->prev = NULL;
-        last->next = NULL;
+        value.next = (first);
```

Comment:
re creating an instance of a new droneRecord class... 
```
DroneRecord* current = first; 
DroneRecord* newDrone = new DroneRecord();
newDrone->droneID =  value.droneID;
newDrone->yearBought = value.yearBought;
newDrone->range = value.range;
		
first = newDrone;
newDrone->next = current;
newDrone->prev = NULL;
current->prev = newDrone;
``` 
Does this make sense? That way it's a bit easier to manipulate and we don't need the if statement?